src/Styleguide/Components/Lightbox/Lightbox.tsx

```diff
@@ -0,0 +1,286 @@
+import { color, space } from "@artsy/palette"
+import FadeTransition from "Components/Animation/FadeTransition"
+import { bind, once, throttle } from "lodash"
+import React from "react"
+import ReactDOM from "react-dom"
+import styled from "styled-components"
+import { Box } from "Styleguide/Elements/Box"
+import { Flex } from "Styleguide/Elements/Flex"
+import { CloseButton } from "./CloseButton"
+import { Slider, SliderProps } from "./LightboxSlider"
+
+const KEYBOARD_EVENT = "keyup"
+const ZOOM_PER_CLICK = 1.4
+const HIDE_ZOOM_SLIDER_AFTER = 2500
+
+const DeepZoomContainer = styled.div`
+  position: fixed !important;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 1000;
+  background-color: ${color("black100")};
+`
+
+export interface LightboxProps {
+  deepZoom: {
+    Image: {
+      xmlns: string
+      Url: string
+      Format: string
+      Overlap: number
+      TileSize: number
+      Size: {
+        Width: number
+        Height: number
+      }
+    }
+  }
+}
+
+export interface LightboxState {
+  shown: boolean
+  element: Element
+  viewer: any
+  deepZoomRef: any
+  slider: SliderProps
+  showZoomSlider: boolean
+  promisedDragon: Promise<any>
+  activityTimer?: NodeJS.Timer
+}
+
+export class Lightbox extends React.Component<LightboxProps, LightboxState> {
+  state = {
+    element: null,
+    viewer: null,
+    shown: false,
+    activityTimer: null,
+    showZoomSlider: true,
+    deepZoomRef: React.createRef(),
+    slider: {
+      min: 0,
+      max: 1,
+      step: 0.001,
+      value: 0,
+    },
+    /**
+     * FIXME: convert to import('openseadragon) once force supports it
+     */
+    promisedDragon: Promise.resolve(require("openseadragon")),
+  }
+
+  renderLightbox() {
+    const { slider } = this.state
+    return (
+      <FadeTransition
+        in={this.state.shown}
+        mountOnEnter
+        unmountOnExit
+        timeout={{ enter: 250, exit: 300 }}
+      >
+        <DeepZoomContainer
+          onMouseMove={this.detectActivity}
+          onWheel={this.detectActivity}
+          onTouchStart={this.detectActivity}
+          onTouchMove={this.detectActivity}
+          innerRef={this.state.deepZoomRef as any}
+        >
+          <Box
+            position="absolute"
+            top={space(3) / 2}
+            right={space(3) / 2}
+            zIndex={1001}
+          >
+            <CloseButton onClick={() => this.hide()} />
```

Comment:
Tiny nit: this could just be `<CloseButton onClick={this.hide} />` instead of creating a new closure.